mlflow/tracking/fluent.py

```diff
@@ -3054,6 +3055,18 @@ def setup_autologging(module):
 
 
 _ACTIVE_MODEL_ID = ContextVar("active_model_id", default=None)
+_ACTIVE_MODEL_ID_STATE = ContextVar("active_model_id_state", default=None)
```

Comment:
Can we combine this as one context var?

Reply:
One caveat of using one context var is that we always need to regenerate the var if we only need to modify one field of the variable. For example:
```
@dataclass
class ActiveModelContext:
    model_id: Optional[str] = None
    source: Optional[ActiveModelIdSource] = None


_ACTIVE_MODEL_CONTEXT = ContextVar(
    "active_model_context",
    default=ActiveModelContext(),
)
```
Then when we need to update source, we need to use `_ACTIVE_MODEL_CONTEXT.set(ActiveModelContext(model_id=existing_model_id, source=new_source)` or ` amc = _ACTIVE_MODEL_CONTEXT.get() + amc.source=new_source` which is not convenient 🤔

Reply:
Actually `amc = _ACTIVE_MODEL_CONTEXT.get() + amc.source=new_source` is probably fine, let me update to use one variable